kasa/smart/modules/triggerlogs.py

```diff
@@ -2,19 +2,22 @@
 
 from __future__ import annotations
 
-from datetime import datetime
+from dataclasses import dataclass
+from typing import Annotated
 
-from pydantic.v1 import BaseModel, Field, parse_obj_as
+from mashumaro import DataClassDictMixin
+from mashumaro.types import Alias
 
 from ..smartmodule import SmartModule
 
 
-class LogEntry(BaseModel):
+@dataclass
+class LogEntry(DataClassDictMixin):
     """Presentation of a single log entry."""
 
     id: int
-    event_id: str = Field(alias="eventId")
-    timestamp: datetime
+    event_id: Annotated[str, Alias("eventId")]
+    timestamp: int
```

Comment:
Note the change from `datetime` to `int` as the timestamp is an `int`.

Reply:
Here's what it shows to me on the water leak sensor with the current master branch, so datetime is actually correct:
```
$ kasa --host 192.168.xx device logs --child asdf
id=114 event_id='d595356d-4953-5654-d59d-b92b6aca9ab2' timestamp=datetime.datetime(2024, 10, 9, 13, 31, 57, tzinfo=datetime.timezone.utc) event='waterDry'
id=113 event_id='c43fc234-4ff2-ac03-d4bf-0254ff2ac03d' timestamp=datetime.datetime(2024, 10, 9, 13, 31, 54, tzinfo=datetime.timezone.utc) event='waterLeak'
id=112 event_id='3e68c39e-b027-e405-7d41-d714fd81bfa8' timestamp=datetime.datetime(2024, 10, 9, 10, 52, 9, tzinfo=datetime.timezone.utc) event='waterDry'
id=111 event_id='0e8743a9-d46a-bdde-67bb-d562b9542219' timestamp=datetime.datetime(2024, 10, 9, 10, 52, 3, tzinfo=datetime.timezone.utc) event='waterLeak'
```

I'm not sure how to tell mashumaro to handle this though, and we should probably also handle the timezone correctly (at some point, so worth adding a comment on that).

Reply:
I think it would be better to keep as an `int` so there's no magic timezone unaware conversion going on, then we can add a cached property that get's the timezone from the device and does a proper conversion to a timezone aware datetime. What do you think?